### PR TITLE
#2163 Add Basic HTTP Authentication (encrypted) support for Open Distro

### PR DESCRIPTION
**Pull request by kdckrs**
Adds support for Basic HTTP Authentication using encoded/encrypted Authorization header. (Open Distro Basic HTTP Authentication requires this when you use the internal users authentication functionality.)

Implements https://github.com/Smile-SA/elasticsuite/issues/2163

### CODE REVIEW

File: src/module-elasticsuite-core/Client/ClientBuilder.php [hunk 1]
```diff
@@ -43,6 +43,7 @@ class ClientBuilder
         'enable_http_auth'      => false,
         'http_auth_user'        => null,
         'http_auth_pwd'         => null,
+        'http_auth_encrypted'   => null,
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Client/ClientBuilder.php:46]: `http_auth_encrypted` here but `http_auth_encoded` below.

File: src/module-elasticsuite-core/Client/ClientBuilder.php [hunk 2]
```diff
@@ -101,6 +102,11 @@ public function build($options = [])
             $clientBuilder->setHandler($handler);
         }
 
+        if (!empty($options['http_auth_user']) && !empty($options['http_auth_pwd']) && $options['http_auth_encoded']) {
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Client/ClientBuilder.php:105]: `http_auth_encoded` here but `http_auth_encrypted` above.

**Review by romainruaud — CHANGES_REQUESTED** [review]
Could you please also add an adminhtml field (Yes/No) to specify if encrypted authentication should be use ?

**Comment by kdckrs** [comment]
> Could you please also add an adminhtml field (Yes/No) to specify if encrypted authentication should be use ?

Cheers for reviewing! I have made the corrections and added the ability to configure this in the admin area. Please let me know if any additional changes are required.

Thanks!

**Review by kdckrs — COMMENTED** [review]
Edit: woops double comment

**Comment by romainruaud** [comment]
Great, thank you for updating this.

I cannot test it by myself (no Open Distro environment with password protection), but I guess you tested it prior to pushing ?

Regards

**Comment by kdckrs** [comment]
> Great, thank you for updating this.
> 
> I cannot test it by myself (no Open Distro environment with password protection), but I guess you tested it prior to pushing ?
> 
> Regards

Yes I have tested it before pushing. If you want we can arrange a quick call to go over it?

**Pull request merged**